collections/src/main/scala/strawman/collection/immutable/WithDefaultImmutableMapOps.scala

```diff
@@ -0,0 +1,24 @@
+package strawman.collection.immutable
+
+import strawman.collection.{IterableFactoryLike, MapFactory}
+import scala.{Option}
+
+trait WithDefaultImmutableMapOps[K, +V, +C <: Map[K, V]] { self: C =>
```

Comment:
Could you make it `private[collection]`?

Reply:
Yes.

Reply:
Made it `private[collection]` and moved it into collection package instead of `immutable`.

Reply:
Instead of `private[collection]`, making `private[immutable]`. As it can be only used for `immutable` collections. For mutable ones will have to define seperate trait, as return types are different. E.g. return type of mapFactory for immutable is `mapFactory[Immutable.Map]` whereas for mutable it is `mapFactory[mutable.Map]`.